### PR TITLE
Fix #310 - Avoid KeyError.

### LINKED ISSUE #310: KeyError with bad request (content type)
Using this 
```python
self.session.put(bucket_url,
                             {'data': {}},
                             auth=self.auth,
                             headers={'If-None-Match': '*'})
```

I had this:
```
2015-06-15 12:23:22,974 INFO  [venusian][Dummy-5] "PUT   /v1/buckets/43904d5519954d59a374660a947d0922" 400 (6 ms) request.summary lang=None; uid=basicauth_0062631f0e1551bde4e4fc5977188122e17253f6517e4cb95528f313ea92aaa6; errno=107; agent=python-requests/2.7.0 CPython/2.7.9 Linux/3.19.0-20-generic; authn_type=BasicAuth; time=2015-06-15T12:23:22; collection_id=bucket; collection_timestamp=1434363802970
2015-06-15 12:23:22,975 ERROR [venusian][Dummy-2] "      " ? (? ms) '/{path:(?!v1).*}' lang=None; exception=Traceback (most recent call last):
  File "/home/mathieu/Code/Mozilla/kinto/.venv/local/lib/python2.7/site-packages/pyramid/tweens.py", line 21, in excview_tween
    response = handler(request)
  File "/home/mathieu/Code/Mozilla/kinto/.venv/local/lib/python2.7/site-packages/pyramid/router.py", line 118, in handle_request
    root = root_factory(request)
  File "/home/mathieu/Code/Mozilla/cliquet/cliquet/authorization.py", line 48, in __init__
    service = utils.current_service(request)
  File "/home/mathieu/Code/Mozilla/cliquet/cliquet/utils.py", line 126, in current_service
    service = services[pattern]
KeyError: '/{path:(?!v1).*}'; uid=None; errno=None; agent=None; authn_type=None
```

Could fix from client with right content type:
```python
            self.session.put(bucket_url,
                             data=json.dumps({'data': {}}),
                             auth=self.auth,
                             headers={'If-None-Match': '*',
                                      'Content-Type': 'application/json'})
```

### PR DESCRIPTION
**Pull request by Natim**

### CODE REVIEW

File: cliquet/tests/test_utils.py [hunk 1]
```diff
@@ -83,3 +84,24 @@ def test_return_env_name_as_uppercase(self):
     def test_return_env_value_is_coerced_to_python(self):
         os.environ.setdefault('CLIQUET_CONF_NAME', '3.14')
         self.assertEqual(read_env('cliquet-conf.name', 12), 3.14)
+
+
+class CurrentServiceTest(unittest.TestCase):
+
+    def test_urrent_service_returns_the_service_for_existing_patterns(self):
```
Comment (ametaireau) [inline cliquet/tests/test_utils.py]: current :)

File: cliquet/tests/test_utils.py [hunk 2]
```diff
@@ -83,3 +84,24 @@ def test_return_env_name_as_uppercase(self):
...
+        request = mock.MagicMock()
+        request.matched_route = mock.MagicMock()
```
Comment (ametaireau) [inline cliquet/tests/test_utils.py]: you don't need to define a new magic mock here.

**Comment by ametaireau** [comment]
Yep, cool!

File: cliquet/utils.py [hunk 3]
```diff
@@ -123,5 +123,9 @@ def current_service(request):
...
         services = request.registry.cornice_services
         pattern = request.matched_route.pattern
-        service = services[pattern]
-        return service
+        try:
+            service = services[pattern]
+        except KeyError:
+            return None
+        else:
+            return service
```
Comment (leplatrem) [inline cliquet/utils.py]: Did you check that there is no usage of current_service() still catching keyerror?

Comment (Natim) [inline cliquet/utils.py]: Yes, I just did :+1:

**Pull request merged**